client/src/planwise/client/scenarios/views.cljs

```diff
@@ -244,53 +257,88 @@
-    (fn [{:keys [name label investment demand-coverage increase-coverage state]} unit-name]
-      [:div
-       [:div {:class-name "section"}
-        [:h1 {:class-name "title-icon"} name]]
-       [edit/scenario-settings @view-state]
-       [:hr]
-       [:div {:class-name "section"}
-        [:h1 {:class-name "large"}
-         [:small (str "Increase in " unit-name " coverage")]
-         (cond
-           (= "pending" state) "loading..."
-           :else               (str increase-coverage " (" (format-percentage increase-coverage @source-demand) ")"))]
-        [:p {:class-name "grey-text"}
-         (cond
-           (= "pending" state) "to a total of"
-           :else               (str "to a total of " (utils/format-number demand-coverage) " (" (format-percentage demand-coverage @source-demand) ")"))]]
-       [:div {:class-name "section"}
-        [:h1 {:class-name "large"}
-         [:small (str "Total " unit-name " under geographic coverage")]
-         (cond
-           (= "pending" state) "loading..."
-           :else  @population-under-coverage)]]
-       [:div {:class-name "section"}
-        [:h1 {:class-name "large"}
-         [:small "Investment required"]
-         "K " (utils/format-number investment)]]
-       [:hr]
-       [edit/create-new-action-component @view-state (or @computing-best-locations? @computing-best-improvements?)]
-       (if (or @computing-best-locations? @computing-best-improvements?)
-         [:div {:class-name "info-computing-best-location"}
-          [:small (if @computing-best-locations?
-                    "Computing best locations ..."
-                    "Computing best improvements...")]])])))
+        population-under-coverage    (subscribe [:scenarios.current/population-under-coverage])
+        providers-from-changeset     (subscribe [:scenarios/providers-from-changeset])]
+    (fn [{:keys [state] :as current-scenario} unit-name error]
+      (if @suggested-locations
+        [:<>
+         [:div {:class-name "suggestion-list"}
+          [edit/create-new-action-component @view-state (or @computing-best-locations? @computing-best-improvements?)]]
+         [suggested-locations-list @suggested-locations state]]
+        [:<>
+         [:div
+          [scenario-info @view-state current-scenario unit-name]
+          [edit/create-new-action-component @view-state (or @computing-best-locations? @computing-best-improvements?)]
+          (if (or @computing-best-locations? @computing-best-improvements?)
+            [:div {:class-name "info-computing-best-location"}
+             [:small (if @computing-best-locations?
+                       "Computing best locations ..."
+                       "Computing best improvements...")]])]
+         [:div (when-not error {:class-name "fade"})]
+         (if error
+           [raise-alert current-scenario error]
+           [changeset/listing-component @providers-from-changeset])
+         [:div (when-not error {:class-name "fade inverted"})]]))))
```

Comment:
Both `div`s make sense only when `error` is falsey, so they can be moved inside the else position in the `if`.